### DISCUSSION TITLE
Let's encrypt http challenge

### DISCUSSION THREAD

**Discussion by l4rth** [coollabsio/coolify#1785] [Q&A]
Hi,

I'm having a hard time setting up coolify, basically because of let's encrypt. I have no dns challenge provider, therefore I need to setup http challenge. For the proxy configuration that's easy:

```yaml
      - '--certificatesresolvers.letsencrypt.acme.httpchallenge=true'
      - "--certificatesresolvers.myresolver.acme.httpchallenge.entrypoint=web"
      - "--certificatesresolvers.myresolver.acme.email=postmaster@example.com"
      - '--certificatesresolvers.letsencrypt.acme.storage=/traefik/acme.json'
```

and labels:
```yaml
      - traefik.http.routers.traefik.tls.certresolver=letsencrypt
      - traefik.http.routers.traefik.tls.domains[0].main=example.com
      - traefik.http.routers.traefik.tls.domains[0].sans=*.example.com
```

this works for my coolify sub-domain, however this already required Traefik-knowledge, which I wanted to avoid by using coolify.

I now tried to setup my first application (gitlab) via the gitlab docker image. After fiddling around, I landed here https://coolify.io/docs/traefik/wildcard-certificates and realised, that I need to tell traefik per application, how to handle the container. 

But I don't have enough traefik knowledge to set it up correctly with the http challenge. I expected the first of the list to work:
```yaml
traefik.enable=true
traefik.http.routers.fowo4wg-0-https.rule=Host(`example.coolify.io`) && PathPrefix(`/`)
traefik.http.routers.fowo4wg-0-https.entryPoints=https
traefik.http.routers.fowo4wg-0-https.middlewares=gzip
traefik.http.routers.fowo4wg-0-https.service=fowo4wg-0-https
traefik.http.services.fowo4wg-0-https.loadbalancer.server.port=80
traefik.http.routers.fowo4wg-0-https.tls=true
traefik.http.routers.fowo4wg-0-https.tls.certresolver=letsencrypt
traefik.http.routers.fowo4wg-0-http.rule=Host(`example.coolify.io`) && PathPrefix(`/`)
traefik.http.routers.fowo4wg-0-http.entryPoints=http
traefik.http.routers.fowo4wg-0-http.middlewares=redirect-to-https
```

but this results in a cert error, and the CN is "TRAEFIK DEFAULT CERT" which is not valid.

I have not enough traefik knowledge to figure out the necessary config, and I also don't know how to debug - which to me currently means, that the effort saved by coolify is zero.

Does someone has done http challenge before, and can help me here?
How can I debug the deployment and especially let's encrypt generation process?

**Comment by Cristy94** [comment]
I have the same issue, where after adding a domain name, the Let's Encrypt certificate is not properly added.

On CapRover, it works, as you can add each domain name individually and it tells immediately after adding it if the certificate could be generated or not.

**Comment by theDanielJLewis** [comment]
Has anyone figured this out?

**Reply by towfiqi** [reply]
I added 3 domains in the domains field. For the 2nd one the letsencrypt SSL was added within 10 minutes. For the 3rd one, it was not being issued even after an hour, so I removed the 2nd domain which didn't actually need it, and clicked the save button and the SSL was issued for the domain that was stuck with Traeffik SSL. You can maybe remove the domain, save the settings, and then add it again to see if the SSL issuance process kicks in.

**Reply by hongqiww** [reply]
Check the proxy log (Servers -> Proxy -> Logs) to see if there is any error there.

**Comment by Slawikus** [comment]
It would be great to have some Docs section on how to setup SSL for the case when domain registrar is not in the list of supported providers, so one cannot use dns challenge. I am also struggling with that.

**Comment by djsisson** [comment]
you cant use http challenge with wildcard certificates or with any kind of proxy dns, lets encrypt has to be able to directly hit traefik without any redirects and traefik needs to pre assign the .wellknown for a named subdomain.
